sdk/src/types/block/output/metadata.rs

```diff
@@ -122,81 +118,88 @@ impl OutputMetadata {
     }
 }
 
-#[allow(missing_docs)]
-pub mod dto {
-    use alloc::string::{String, ToString};
-    use core::str::FromStr;
-
-    use serde::{Deserialize, Serialize};
+#[cfg(feature = "serde")]
+mod dto {
+    use serde::Deserialize;
 
     use super::*;
-    use crate::types::block::Error;
-
-    /// DTO for an [`OutputMetadata`].
-    #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
-    #[serde(rename_all = "camelCase")]
-    pub struct OutputMetadataDto {
-        pub block_id: String,
-        pub transaction_id: String,
-        pub output_index: u16,
-        pub is_spent: bool,
-        #[cfg_attr(feature = "serde", serde(skip_serializing_if = "Option::is_none"))]
-        pub milestone_index_spent: Option<u32>,
-        #[cfg_attr(feature = "serde", serde(skip_serializing_if = "Option::is_none"))]
-        pub milestone_timestamp_spent: Option<u32>,
-        #[cfg_attr(feature = "serde", serde(skip_serializing_if = "Option::is_none"))]
-        pub transaction_id_spent: Option<String>,
-        pub milestone_index_booked: u32,
-        pub milestone_timestamp_booked: u32,
-        pub ledger_index: u32,
-    }
 
-    impl OutputMetadataDto {
-        /// Returns the output id.
-        pub fn output_id(&self) -> Result<OutputId, crate::types::block::Error> {
-            OutputId::new(TransactionId::from_str(&self.transaction_id)?, self.output_index)
-        }
-    }
+    impl<'de> Deserialize<'de> for OutputMetadata {
+        fn deserialize<D: serde::Deserializer<'de>>(d: D) -> Result<Self, D::Error> {
+            #[derive(Deserialize)]
+            #[serde(untagged)]
+            enum OutputMetadataAlias {
```

Comment:
What is this ?

Reply:
I think we need to have one type with output_id but that serialises to something with the field split in 2